core/src/main/java/org/elasticsearch/client/transport/TransportClient.java

```diff
@@ -113,6 +113,9 @@ public Builder addPlugin(Class<? extends Plugin> pluginClass) {
          * Builds a new instance of the transport client.
          */
         public TransportClient build() {
+            if (settings.get("plugin.types") != null) {
+                throw new IllegalArgumentException("plugin.types is no longer supported. Use the addPlugin method on TransportClient.");
```

Comment:
If not throwing when the `settings` are passed to the builder, maybe `IllegalStateException` since `IllegalArgumentException` is really meant for method parameters gone bad?